spec/controllers/gitlab/projects/events_controller_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'rails_helper'
+
+describe Gitlab::Projects::EventsController, type: :controller do
+  let!(:story) { FactoryBot.create(:story, :with_project, branch: 'test-branch') }
+  let!(:project) { story.project }
+  let!(:integration) do
+    FactoryBot.create(
+      :integration,
+      project: project,
+      kind: 'gitlab',
+      data: {
+        'secret_key' => 'secret_key'
+      }
+    )
+  end
+  let(:params) {
+    {
+      'object_attributes' => {
+        'source_branch' => 'test-branch'
+      }
+    }
+  }
+  let(:headers) { { 'X-Gitlab-Token': token } }
+
+  describe '#create' do
+    before do
+      allow(Gitlab::ProjectEventsService).to receive(:perform).and_return(true)
+      allow(subject).to receive(:params).and_return(params)
+    end
+
+    context 'with an valid secret key' do
+      let(:token) { 'secret_key' }
```

Comment:
add this value in the params if the value of secret key changes you gonna need to change in all other places, just add a descriptive name in the let